Fixed showing of velocity of steps.

karmux opened the pull request:
Fixes #2863.

Umcaruje commented on src/tracks/Pattern.cpp:
```diff
@@ -1149,7 +1149,7 @@ void PatternView::paintEvent( QPaintEvent * )
 			{
 				p.drawPixmap( x, y, stepoff );
-				p.setOpacity( vol / 100.0 );
+				p.setOpacity( min(vol, 100) / 100.0 );
```
I do not get this, there is no way to distinguish notes over 100% velocity now:
![screenshot from 2016-08-12 00 04 40](https://cloud.githubusercontent.com/assets/6282045/17606568/692b46f8-6020-11e6-96ea-b506296792eb.png)

Here's a pattern opened in pianoroll, as you see all the steps look the same while they are not the same velocity.

Umcaruje replied:
So I did some testing, and it works really nice if you revert the pixmap to the previous, half-transparent one: 
![screenshot from 2016-08-12 00 55 47](https://cloud.githubusercontent.com/assets/6282045/17607740/fd950bce-6027-11e6-91b9-77a04d63a76d.png)

Now I also did a test without the `if(vol>100)` clause and with `p.setOpacity( vol / 200.0 );`, with your opaque pixmap:
![screenshot from 2016-08-12 00 57 26](https://cloud.githubusercontent.com/assets/6282045/17607792/4d0ca9c8-6028-11e6-9f56-001f3d1f2d60.png)
which did not look as nice because of the linear scale of the opacity, now this could be fixed by using a logscale, but I think we're much better off just drawing pixmaps on top of each other.

karmux commented:
I'm working on it to make 100% volumes bright enough and over 100% volumes distinguishable.

karmux commented:
@Umcaruje and others, does it look better now?

![screenshot_20160814_161738](https://cloud.githubusercontent.com/assets/2575235/17649315/ba094856-623a-11e6-9879-a3c64ad4b6ec.png)

I removed alpha, added alpha back and adjusted alpha to 85% (100% volume). In case of volumes over 100% another pixmap is added using CompositionMode_Plus. Also I removed border from step_btn_on_100 image and now it shows either dark or light border. Previously it was always dark.

sfxr steps has 100% volumes and 9th step of kicker is also 100%.

Would like to get some feedback before I commit this one.

Umcaruje commented:
The pixmap looks a lot lighter now, and I'm not sure I'm liking it. The change looks too excessive to me now.

karmux commented:
Does near 200% volumes look too bright or also 100% volumes?
If you use over 100% volumes much then the overall picture is much brighter now and you probably want all volumes to be darker. I kind of agree with you in that.
On the other hand if you rarely go over 100% then you want 100% to be about as bright as on the screenshot.

I checked in LMMS 1.1.3 and there were no way to say if step is 100% or 200% without looking into piano roll. I guess LMMS hasn't had functionality yet to distinguish volumes over 100%.

tresf commented:
200% note velocity for steps... is that even possible with the mouse wheel?  Sounds like a bug.

karmux commented:
@tresf if you hold mouse on the square and scroll up/down then it changes volume of that step from 0 to 100%. You can scroll down from 200 but not back to 200. In piano roll scrolling volume works both ways between 0 and 200. But this is a different issue.

If I make 200 darker and leave 100 as bright as it is, then there is again issue that volume of notes over 100% is not distinguishable. If I make all volumes darker then again 100% volume seems like not maxed out in 0-100 scale.

tresf commented:
It is only a separate issue if we decide to keep the incorrect behavior.  Two GUI adjustments, same element, different range.

The existing behavior is incorrect (200 looks like 100).   The answer depends on the intended behavior.

karmux commented:
Opacity of image 50, instead of 80 looks like this:
![screenshot_20160815_015613](https://cloud.githubusercontent.com/assets/2575235/17652692/f7f7c178-628b-11e6-8b31-ba4cf1e460f8.png)
I don't like this as 100% volumes are too dark.

For me scoll limit at 100% is a good thing and I'm not taking it as a bug. I never use volumes over 100% in beats but maybe it's just me.

As LMMS has had this scroll limit at 100% for a very long time and previously there were no visual difference between 100% and 200% beat volumes, then perhaps we keep it like this and just fix velocity between 0-100%?

jasp00 commented:
Have you tried `sqrt(vol / 200.0)`?

tresf commented:
> For me scoll limit at 100% is a good thing and I'm not taking it as a bug. I never use volumes over 100% in beats but maybe it's just me.

Right, which makes your original proposal better.  To NOT visually distinguish 100% - 200% is bad UI, so to adjust your PR to hide a 200% note is bad UI.

karmux commented:
Using `sqrt(vol / 200.0)` and no image opacity, it looks pretty good :thumbsup:

![screenshot_20160815_210259](https://cloud.githubusercontent.com/assets/2575235/17673857/afac74ae-632b-11e6-8f0c-659c3e17ae26.png)

Umcaruje commented:
Works good, though I'd prefer if you used the original icon, just without transparency: ![step_btn_on_100](https://cloud.githubusercontent.com/assets/6282045/17792910/0468dbc0-65a4-11e6-8468-ebf88b2c48ca.png)
![screenshot from 2016-08-19 00 27 13](https://cloud.githubusercontent.com/assets/6282045/17792913/09b395d4-65a4-11e6-904f-cc72cd8a8a7c.png)

karmux commented:
Original image without transparency looks blurry.

![screenshot_20160820_122459](https://cloud.githubusercontent.com/assets/2575235/17830333/25a72eda-66d1-11e6-8c5b-0fca552a9788.png)

Umcaruje commented:
... that looks a lot different than my screenshot

karmux commented:
@Umcaruje it seems I removed transparency differently. I've committed your image now. How did you exactly removed transparency from original image? I used GIMP and followed [this](https://docs.gimp.org/ca/gimp-layer-alpha-remove.html).

Umcaruje commented:
I just stacked duplicates of the image until there was no more transparent pixels. Removing the alpha channel would be the same as if you added a fully white layer under the image. Stacking duplicates is like reverse-engineering the image, so it looks the same before its opacity was dropped down.

RebeccaDeField commented:
@karmux @Umcaruje  Here is the original source file exported without any transparency.

![step](https://cloud.githubusercontent.com/assets/7960169/18605783/d6f93fa4-7c50-11e6-944d-cbb3c85e8d88.png)

karmux commented:
@RebeccaDeField Umcarruje's image is already committed to this pull request. If they are identical then I don't change anything.

RebeccaDeField commented:
@karmux Yes, that is what I was double checking. The file looks good to me :)

Umcaruje commented:
Hey @karmux can you fix the conflicts so we can merge this one

Umcaruje commented:
Tested this again, works good. The code looks good too. Merging.

midi-pascal commented:
@Umcaruje The last merge deleted step_btn_on_yellow.png from default theme.
Now make install crash because of this.

Umcaruje commented:
>Now make install crash because of this.

Did you try doing a clean compile? (removing the build folder and the target folder, then compiling again)

midi-pascal commented:
@Umcaruje :+1:
